test/e2e/storage/non_graceful_node_shutdown_scale.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"github.com/onsi/ginkgo/v2"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"k8s.io/kubernetes/test/e2e/storage/drivers"
+	storageframework "k8s.io/kubernetes/test/e2e/storage/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+/*
+The Goal of this test is to do stress testing on NodeOutOfServiceVolumeDetach feature
+
+This test assumes the following:
+- The infra is GCP.
+- Number of nodes in the Kubernetes cluster is at least 10, otherwise the test will be skipped.
+- NodeOutOfServiceVolumeDetach feature is enabled.
+
+This test performs the following:
+ 1. Deploys a gce-pd csi driver.
+ 2. Creates a gce-pd csi storage class.
+ 3. Taints 50% of nodes out of 'N' nodes.
+ 4. Creates a stateful set with "N/2" number of replicas.
```

Comment:
With current implementation the number of pods that we can create for this test is always limited by the number of nodes because sts replica pods spreads across the nodes. 
For example, for '10' nodes, STS can only have 5 replica pods and then the nodes where these pods got scheduled can be shutdown to move it to other '5' healthy nodes. 

While I was working on this, one idea that came was the following:
- We have let us say 'N' number of nodes in the k8s cluster. 
- Create 'K' statefulsets with only 1 or 2 replica count ( Let us say replica count is represented by 'RC'). 
- Now have at least 'RC' number of nodes available ( these are nodes that do not go through shutdown in the test )

This way number of pods will not be constrained by the number of nodes. 

cc @xing-yang